android/src/toga_android/fonts.py

```diff
@@ -21,31 +22,25 @@ class Font:
     def __init__(self, interface):
         self.interface = interface
 
-    def get_size(self):
-        # Default system font size on Android is 14sp (sp = dp, but is separately
-        # scalable in user settings). For what it's worth, Toga's default is 12pt.
-        if self.interface.size == SYSTEM_DEFAULT_FONT_SIZE:
-            font_size = 14
-        else:
-            font_size = self.interface.size
-        return float(font_size)
+    def apply(self, tv, default_size, default_typeface):
+        """Apply the font to the given native widget.
 
-    def get_style(self):
-        if self.interface.weight == BOLD:
-            if self.interface.style == ITALIC:
-                return Typeface.BOLD_ITALIC
-            else:
-                return Typeface.BOLD
-        elif self.interface.style == ITALIC:
-            return Typeface.ITALIC
+        :param tv: A native instance of TextView, or one of its subclasses.
+        """
+        if self.interface.size == SYSTEM_DEFAULT_FONT_SIZE:
+            tv.setTextSize(TypedValue.COMPLEX_UNIT_PX, default_size)
         else:
-            return Typeface.NORMAL
+            # The default size for most widgets is 14sp, so mapping 1 Toga "point" to 1sp
+            # will give relative sizes that are consistent with desktop platforms. It also
+            # means font sizes will all change proportionately if the user adjusts the
+            # text size in the system settings.
+            tv.setTextSize(TypedValue.COMPLEX_UNIT_SP, self.interface.size)
 
-    def get_typeface(self):
+        cache_key = (self.interface, default_typeface)
```

Comment:
Why is the default typeface part of the cache key?

Reply:
Because it may affect the resulting font when font_family is `SYSTEM`. The default font on most widgets is `sans-serif`, but on buttons it's `sans-serif-medium`, which is intermediate between normal and bold.